$MirOS: ports/net/djbdns/patches/patch-src_taia_now_c,v 1.1.201.1 2005/05/14 23:21:05 tg Exp $
--- src/taia_now.c.orig	Sun Feb 11 21:11:23 2001
+++ src/taia_now.c	Fri Feb 23 01:02:57 2007
@@ -1,12 +1,26 @@
+#include <sys/param.h>
+#if defined(MirBSD) && (MirBSD > 0x09AB)
+#include <sys/taitime.h>
+#endif
 #include <sys/types.h>
 #include <sys/time.h>
 #include "taia.h"
 
 void taia_now(struct taia *t)
 {
+#ifndef __TAI64_BIAS
   struct timeval now;
+
   gettimeofday(&now,(struct timezone *) 0);
   tai_unix(&t->sec,now.tv_sec);
   t->nano = 1000 * now.tv_usec + 500;
   t->atto = 0;
+#else
+  tai64na_t v;
+
+  t->atto = 0;
+  taina_time(&v);
+  t->sec.x = v.secs;
+  t->nano = v.nano;
+#endif /* __TAI64_BIAS */
 }
